### ISSUE TITLE
Streams: Stream for multiple timelines

### ISSUE THREAD

**Issue by kkostov** [TootSDK/TootSDK#308]
We can imagine a facility where you get a swift stream from multiple timelines e.g. 

```swift
let stream = try! await client.streaming.subscribe(to: [.userNotification, .public])

do {
    // listen for events
    for try await event in stream {
        print("got event")
        switch event {
        case .connectionUp(let timeline):
            //...
        case .connectionDown(let timeline):
            //...
        case .receivedEvent(let timeline, let eventContent):
            //...
        }
    }
} catch {
    print(String(describing: error))
}
``` 

Under the hood though, it will always be a socket per timeline.

**Comment by daprice** [comment]
This should already be doable by subscribing to the stream for each one and then calling AsyncAlgorithms’ [merge](https://github.com/apple/swift-async-algorithms/blob/main/Sources/AsyncAlgorithms/AsyncAlgorithms.docc/Guides/Merge.md) with the streams, unless I’m mistaken?

The implementation we have now does share a single WebSocket for all subscriptions that use the same `StreamingClient`. If for some reason you want a socket per timeline you can make more StreamingClient instances or use `TootClient.beginStreaming`.

**Comment by helje5** [comment]
Makes sense. Given that it is a single stream in the source, and the specific timeline stream isn't typed out, i.e. the same for all timelines, it may still make sense to provide one stream subscribing multiple topics from the beginning.
Vs creating 10 streams targeting the same socket and then just merge them again.

**Comment by kkostov** [comment]
I think the socket split would depend largely on flavour support - can one actually use the same socket for multiple subscriptions in every flavour? Where this is not the case, we can handle it in code so a consuming app doesn't have to know exactly how many sockets there are unless it's setup manually like @daprice mentions above.

We have to be mindful that for "normal users" of fedi these won't contain a lot of notifications, but "very large prominent users" will be blasting significant amount of updates from each timeline. 
There is some sense in managing this on a per-timeline basis depending on what the user is currently doing in the app.

**Comment by daprice** [comment]
@kkostov did you mean to close this? Even though you can use `merge` to get the same effect for now, I’d still like to refactor the streaming subscriber code to take the feedback from this thread into account at some point
